### ISSUE TITLE
Support log delivery to S3 for wafv2 web_acl_logging_configuration

### ISSUE THREAD

**Issue by acidix** [hashicorp/terraform-provider-aws#22085]
## Community Note
Please vote on this issue by adding a 👍 reaction to the original issue to help the community and maintainers prioritize this request
Please do not leave "+1" or other comments that do not add relevant new information or questions, they generate extra noise for issue followers and do not help prioritize the request
If you are interested in working on this issue or have submitted a pull request, please leave a comment

## Description
The "web_acl_logging_configuration" currently only supports a Kinesis Firehose Stream as a destination for log delivery. As of December 6, 2021 AWS is supporting to send logs to S3 buckets directly (https://aws.amazon.com/de/about-aws/whats-new/2021/12/awf-waf-cloudwatch-log-s3-bucket/). This functionality should be implemented in the terraform provider.

## New or Affected Resource(s)
web_acl_logging_configuration

Potential Terraform Configuration
```hcl
resource "aws_wafv2_web_acl_logging_configuration" "example" {
  log_destination_configs = [aws_s3_bucket.example.arn]
  resource_arn            = aws_wafv2_web_acl.example.arn
  redacted_fields {
    single_header {
      name = "user-agent"
    }
  }
}
```

**Comment by acidix** [comment]
Use of a CloudWatch log group seems to be supported as well. See https://pkg.go.dev/github.com/aws/aws-sdk-go-v2/service/wafv2#Client.PutLoggingConfiguration for details.

**Comment by iokiwi** [comment]
Yep can this issue also either reflect that CloudWatch Log should be supported or link to the issue for supporting CloudWatch Log

**Comment by sjonpaulbrown** [comment]
@acidix With the existing resource, have you tried using an S3 bucket arn as a log destination?

This functionality already exists within the resource. While the resource documentation may need to be updated, the resource seems to support this functionality. I have tested this out with a CloudWatch log group, and it works as expected. 

The proper naming conventions have to be used for WAF to be able to successfully deliver the logs. In my case, when the log group name did not match the naming convention, it failed.

For CloudWatch logs, the log group name must begin with the following prefix ` aws-waf-logs- `, and for S3 buckets, the bucket name must start with the following prefix `aws-waf-logs-`. According to the documentation, if the bucket resides within the same account, the WAF service will handle the addition of the proper bucket policy. Otherwise, the policy has to be explicitly defined on the bucket so that the WAF logs can be delivered to a bucket in a separate account.

All that being said, I believe this is already supported within the existing resource and code. There is not any additional validation that is taking place. The `log_destination_config` [just expects the items in the array to be valid arns](https://github.com/hashicorp/terraform-provider-aws/blob/187f1659a4fef11ac314567273b5470afe6b662f/internal/service/wafv2/web_acl_logging_configuration.go#L40). 

I would read the [documentation](https://docs.aws.amazon.com/waf/latest/developerguide/logging-cw-logs.html) to ensure your bucket is properly configured, and provide the S3 bucket arn as a destination in `log_destination_config`.  

Documentation Links:
* https://docs.aws.amazon.com/waf/latest/developerguide/logging-cw-logs.html
* https://docs.aws.amazon.com/waf/latest/developerguide/logging-s3.html

**Comment by poflynn** [comment]
Thanks so much @sjonpaulbrown ! Indeed the code below worked for me.

`resource "aws_wafv2_web_acl_logging_configuration" "this" {
  log_destination_configs = [aws_s3_bucket.waf-logs.arn]
  resource_arn            = module.alb_wafv2.web_acl_id
}`
